Jenkinsfile

```diff
@@ -62,39 +87,52 @@ for(b = 0; b < builderNodes.size(); b++) {
-                        if(statusCode != 0) {
-                            setBuildStatus("Building examples", "FAILURE");
-                        }
-
-                        // Parse test output for GCC warnings
-                        // **NOTE** driving WarningsPublisher from pipeline is entirely undocumented
-                        // this is based mostly on examples here https://github.com/kitconcept/jenkins-pipeline-examples
-                        // **YUCK** fatal errors aren't detected by the 'GNU Make + GNU C Compiler (gcc)' parser
-                        // however JENKINS-18081 fixes this for
-                        // the 'GNU compiler 4 (gcc)' parser at the expense of it not detecting make errors...
-                        def parserName = ("mac" in nodeLabel) ? "Apple LLVM Compiler (Clang)" : "GNU compiler 4 (gcc)";
-                        step([$class: "WarningsPublisher",
-                            parserConfigurations: [[parserName: parserName, pattern: uniqueMsg]],
-                            unstableTotalAll: '0', usePreviousBuildAsReference: true]);
-
-                        // Archive output
-                        archive uniqueMsg;
+            stage("Downloading and building GeNN (" + env.NODE_NAME + ")") {
+                sh 'bin/download_and_build_genn.sh'
+            }
+
+            runBuild("examples", nodeLabel);
+            runBuild("projects", nodeLabel);
+            runBuild("tools", nodeLabel);
+            runBuild("tests", nodeLabel);
+
+            // Parse test output for GCC warnings
+            recordIssues enabledForFailure: true, tool: gcc(pattern: "**/msg_build_*_" + env.NODE_NAME, id: "gcc_" + env.NODE_NAME)
+
+            stage("Running tests (" + env.NODE_NAME + ")") {
+                setBuildStatus("Running tests (" + env.NODE_NAME + ")", "PENDING");
+                dir("tests") {
+                    // Generate unique name for message
+                    def uniqueMsg = "msg_test_results_" + env.NODE_NAME;
+                    def runTestsCommand = "./tests --gtest_output=xml:gtest_test_results.xml 1>> \"" + uniqueMsg + "\" 2>> \"" + uniqueMsg + "\"";
+                    def runTestsStatus = sh script:runTestsCommand, returnStatus:true;
+
+                    // Archive output
+                    archive uniqueMsg;
+
+                    // If tests failed, set failure status
+                    if(runTestsStatus != 0) {
+                        setBuildStatus("Running tests (" + env.NODE_NAME + ")", "FAILURE");
+                    }
+                }
+            }
+
+            stage("Gathering test results (" + env.NODE_NAME + ")") {
+                junit '**/*_test_results.xml'
+            }
+
+            // Jamie's machine doesn't have clang-tidy installed, but we probably don't need to run this on multiple machines anyhow...
+            if(env.NODE_NAME == "INF900802") {
```

Comment:
you could make this a little nicer by testing:
```java
if("clang_tidy" in nodeLabel) {
```
(I added the clang_tidy tag to your machine on Jenkins)